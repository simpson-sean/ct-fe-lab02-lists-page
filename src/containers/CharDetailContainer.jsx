import React, { Component } from 'react';
import { findCharacterById } from '../services/RickAndMortyApi';
import { NavLink } from 'react-router-dom';


export default class CharDetailContainer extends Component {

    state = {
        characterDetails: {},
        loading: true,
    };

    componentDidMount() {
        findCharacterById(this.props.match.params.id).then((character) =>
        this.setState({ characterDetails: character, loading: false }));
    };

    render() {
        return (
            <div data-testid="testing-div">
                <h1>Character Details</h1>
                <center>
                    <img src={this.state.characterDetails.image} id='card'></img>
                </center>
                <h2>Name: {this.state.characterDetails.name}</h2>
                <h2>Status: {this.state.characterDetails.status}</h2>
                <NavLink className='nav-links' exact to="/">Home</NavLink>
            </div>
        )
    }
}