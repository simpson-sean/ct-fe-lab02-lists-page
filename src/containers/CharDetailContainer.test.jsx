import React from 'react';
import { render, screen } from '@testing-library/react';
import CharDetailContainer from './CharDetailContainer';


describe('CharDetailContainer', () => {
    it('renders a single character', () => {
        render(
         <CharDetailContainer />
        );

        const test = screen.getByTestId('testing-div')

        expect(test).toEqual("testing-div");
    });
})