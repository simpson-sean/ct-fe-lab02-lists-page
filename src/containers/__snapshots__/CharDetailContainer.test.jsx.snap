// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharDetailContainer renders a single character 1`] = `
<div
  data-testid="testing-div"
>
  <h1>
    Character Details
  </h1>
  <center>
    <img
      id="card"
    />
  </center>
  <h2>
    Name: 
  </h2>
  <h2>
    Status: 
  </h2>
  <a
    aria-current="page"
    class="nav-links active"
    href="/"
  >
    Home
  </a>
</div>
`;
